### DISCUSSION TITLE
Code modification about moisture and salinity

### DISCUSSION THREAD

**Discussion by liyunfengcaas** [KUL-RSDA/AquaCrop#324] [Q&A] [answered]
We intend to introduce daily moisture (cm3/cm3) and salinity (ECe) data from outside sources into the source code, so we want to know how the daily moisture and salinity of AquaCrop changes while the program is running. And where should we make the changes in AquaCrop.

**Comment by lbusschaert** [comment, marked as answer]
Dear Yufeng,

At the end of each day of the simulation, Aquacrop provides the simulation results for the soil water and soil salinity content (ECe) in each of the 12 compartments (they can be requested as output – in the AquaCrop version with the graphical user interface the output is available in the CompWC.OUT and CompEC.OUT files). By adjusting the value of the 12 state variables of the Soil water content and the 12 state variables of the ECe at the end of the day and before the start of the simulation of the next day, AquaCrop will update the soil water and soil salinity stress and runs the simulation with the adjusted stresses. To evaluate the way of doing, just run an example the AquaCrop with the graphical user interface, and update after a specific date the soil water content (click on update button in the run menu), and observe the changes in the simulation.

Best regards
